js/react_native/lib/backend.ts

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {Buffer} from 'buffer';
+import {Backend, InferenceSession, SessionHandler, Tensor,} from 'onnxruntime-common';
+
+import {binding, Binding} from './binding';
+
+type SupportedTypedArray = Exclude<Tensor.DataType, string[]>;
+
+const tensorTypeToTypedArray = (type: Tensor.Type):|Float32ArrayConstructor|Int8ArrayConstructor|Int16ArrayConstructor|
+    Int32ArrayConstructor|BigInt64ArrayConstructor|Float64ArrayConstructor => {
+      switch (type) {
+        case 'float32':
+          return Float32Array;
+        case 'int8':
+          return Int8Array;
+        case 'int16':
+          return Int16Array;
+        case 'int32':
+          return Int32Array;
+        case 'bool':
+          return Int8Array;
+        case 'float64':
+          return Float64Array;
+        case 'int64':
+          /* global BigInt64Array */
+          /* eslint no-undef: ["error", { "typeof": true }] */
+          return BigInt64Array;
+        default:
+          throw new Error(`unsupported type: ${type}`);
+      }
+    };
+
+class OnnxruntimeSessionHandler implements SessionHandler {
+  #inferenceSession: Binding.InferenceSession;
+  #key: string;
+
+  inputNames: string[];
+  outputNames: string[];
+
+  constructor(path: string) {
+    this.#inferenceSession = binding;
+    this.#key = path;
+    this.inputNames = [];
+    this.outputNames = [];
+  }
+
+  async loadModel(options: InferenceSession.SessionOptions): Promise<void> {
+    try {
+      // load a model
+      const results: Binding.ModelLoadInfoType = await this.#inferenceSession.loadModel(this.#key, options);
+      // resolve promise if onnxruntime session is successfully created
+      if (results.key !== this.#key) {
+        throw new Error('Session key is invalid');
+      }
+
+      this.inputNames = results.inputNames;
+      this.outputNames = results.outputNames;
+      return Promise.resolve();
+    } catch (e) {
+      throw new Error('Can\'t load a model');
+    }
+  }
+
+  async dispose(): Promise<void> {
+    return Promise.resolve();
+  }
+
+  startProfiling(): void {
+    // TODO: implement profiling
+  }
+  endProfiling(): void {
+    // TODO: implement profiling
+  }
+
+  async run(feeds: SessionHandler.FeedsType, fetches: SessionHandler.FetchesType, options: InferenceSession.RunOptions):
+      Promise<SessionHandler.ReturnType> {
+    // eslint-disable-next-line no-async-promise-executor
+    return new Promise(async (resolve, reject) => {
+      try {
+        // Java API doesn't support preallocated output names and allows only string array as parameter.
+        const outputNames: Binding.FetchesType = [];
+        for (const fetch in fetches) {
+          if (Object.prototype.hasOwnProperty.call(fetches, fetch)) {
```

Comment:
you can give additional check here, `if (fetches[fetch]) { throw new Error('Java API doesn't support preallocated output and allows only names as string array as parameter.'); }`